src/assets/TokenRatesController.ts

```diff
@@ -169,21 +308,31 @@ export class TokenRatesController extends BaseController<
    * @returns Promise resolving when this operation completes
    */
   async updateExchangeRates() {
-    if (this.tokenList.length === 0) {
+    if (this.tokenList.length === 0 || this.disabled) {
       return;
     }
-    const newContractExchangeRates: { [address: string]: number } = {};
     const { nativeCurrency } = this.config;
-    const pairs = this.tokenList.map((token) => token.address).join(',');
-    const query = `contract_addresses=${pairs}&vs_currencies=${nativeCurrency.toLowerCase()}`;
-    const prices = await this.fetchExchangeRate(query);
-    this.tokenList.forEach((token) => {
-      const address = toChecksumHexAddress(token.address);
-      const price = prices[token.address.toLowerCase()];
-      newContractExchangeRates[address] = price
-        ? price[nativeCurrency.toLowerCase()]
-        : 0;
-    });
+
+    const slug = await this.getChainSlug();
+
+    const newContractExchangeRates: ContractExchangeRates = {};
+    if (!slug) {
+      this.tokenList.forEach((token) => {
+        const address = toChecksumHexAddress(token.address);
+        newContractExchangeRates[address] = undefined;
+      });
+    } else {
+      const pairs = this.tokenList.map((token) => token.address).join(',');
```

Comment:
[nit] @wachunei I know you didn't name this const but it seems misnamed to me? It's a list of tokens that are each one side of a pair with the native currency right?

Reply:
No, this is just a concatenation of token addresses joined by a comma, from something like `[{ address: '0x1...', ...restOfTokenObject1 }, { address: '0x2...', ...restOfTokenObject2 }]` into `"0x1...,0x2..."`, which is what CoinGecko takes in as query param for the API call.

Reply:
I agree is not a great name! But you know, naming stuff is hard hahahaha

Reply:
`pairs` should be `addresses`, sorry for necroposting